lib/miq_expression/field.rb

```diff
@@ -26,7 +26,10 @@ def date?
   end
 
   def self.valid_field?(field)
-    FIELD_REGEX.match(field).present? if field.kind_of?(String)
+    if field.kind_of?(String)
+      match = FIELD_REGEX.match(field)
+      match.present? && match[:model_name].safe_constantize.present?
+    end
```

Comment:
implies that non-string fields are invalid?  If so, maybe change guts of methods to be more linear.
```ruby
def self.valid_field?(field)
  return false unless field.kind_of?(String)
  match = FIELD_REGEX.match(field)
  match.present? && match[:model_name].safe_constantize.present?
end
```